### PR TITLE
UCP/INFO: Print UCP context, worker, and endpoint info.

### LINKED ISSUE #761: ucx_info -p is broken

### PR DESCRIPTION
**Pull request by yosefe**
Fixes #761

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2407/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1744/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2411/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1750/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2412/ for details.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1751/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask @shamisp pls review
MellanoxLab failure is unrelated

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by shamisp** [comment]
@yosefe please respond on the above comments.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1758/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1759/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
bot:ornl:retest

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1768/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp can you pls review?

**Comment by shamisp** [comment]
@yosefe I reviewed - waiting for your response.

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -645,18 +660,15 @@ void ucp_worker_destroy(ucp_worker_h worker);
...
  *
- * @param [in] worker       Worker object whose address to return.
+ * @param [in] worker       Worker object ro print information for.
  * @param [in] stream       Output stream to print the information to.
- * @param [in] title        Configuration title to print.
- * @param [in] print_flags  Flags that control various printing options.
  */
-void ucp_worker_proto_print(ucp_worker_h worker, FILE *stream, const char *title,
-                            ucs_config_print_flags_t print_flags);
+void ucp_worker_print_info(ucp_worker_h worker, FILE *stream);
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: yes

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -645,18 +660,15 @@ void ucp_worker_destroy(ucp_worker_h worker);
...
 /**
  * @ingroup UCP_WORKER
- * @brief Print information about protocols that would be used by a worker.
+ * @brief Print information about the worker.
  *
  * This routine prints information about the protocols being used, thresholds,
  * UCT transport methods, and other useful information associated with the worker.
  *
- * @param [in] worker       Worker object whose address to return.
+ * @param [in] worker       Worker object ro print information for.
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: yep

File: src/ucp/api/ucp.h [hunk 3]
```diff
@@ -598,6 +598,21 @@ void ucp_cleanup(ucp_context_h context_p);
...
+/**
+ * @ingroup UCP_CONTEXT
+ * @brief Print context information.
+ *
+ * This routine prints information about the context configuration, including
+ * memory domains, transport resources, and other useful information associated
+ * with the context.
+ *
+ * @param [in] context      Context object whose configuration to print.
+ * @param [in] stream       Output stream to print the information to.
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: then need to know the size in advance

File: src/ucp/api/ucp.h [hunk 4]
```diff
@@ -888,6 +900,19 @@ ucs_status_ptr_t ucp_disconnect_nb(ucp_ep_h ep);
 
 /**
  * @ingroup UCP_ENDPOINT
+ * @brief Print endpoint information.
+ *
+ * This routine prints information about the endpoint transport methods, limits,
+ * thresholds, and other useful information associated with the endpoint.
```
Comment (yosefe) [inline src/ucp/api/ucp.h]: will remove

File: src/ucp/api/ucp.h (hunk 3 above)
Comment (shamisp) [inline src/ucp/api/ucp.h]: ok

File: src/ucp/api/ucp.h (hunk 1 above)
Comment (shamisp) [inline src/ucp/api/ucp.h]: ok

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2427/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1773/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shamisp i've fixed the doc

**Comment by shamisp** [comment]
👍

**Pull request merged**